put sum when behind ff like sum when template

mkangia opened the pull request:
## Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
Make `Sum When` available on HQ behind a feature flag for development purposes instead of just static reports. 
Using the same feature flag that gives access to `Sum when template`

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
`UCR_SUM_WHEN_TEMPLATES`

## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->
Users can now use [sum when](https://commcare-hq.readthedocs.io/ucr.html?highlight=ucr#sumwhencolumn-and-sumwhentemplatecolumn) columns on HQ Configurable reports UI, just like sum when template columns.

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I am certain that this PR will not introduce a regression for the reasons below

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->

### Safety story
<!--
Describe any other pieces to the safety story including
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->
No domain has this feature flag enabled on prod. It was only used by icds domains on India. So there should be no behavior change for any project outside the ones enabled.

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

mkangia committed:
put sum when behind ff like sum when template

orangejenny reviewed (approved):
Code looks fine. It does make me nervous to have this as a flag, but it's marked custom, and I think as long as we're paying attention to which projects turn this on it'll be all right.

mkangia commented:
@orangejenny happy to add any warnings on the toggle if you suggest? Its only used by limited projects as of now. I believe UCRs itself are used by a limited set.

orangejenny commented:
Maybe warn in the toggle description?

mkangia committed:
add warning

mkangia commented:
@orangejenny added warning.

orangejenny commented:
thanks 🙂

esoergel commented:
When this feature flag is enabled for a domain, who will be able to write these expressions?  Not just superusers, right?  I imagine domain admins also have access (though I'm not sure).  If I recall correctly, this feature allows use of handwritten, unescaped SQL expressions.  When Emord and I tested this, we were able to write a UCR which deleted the postgres users database.

A feature flag limits the number of domains and admin users which will have this capability, but it does not safeguard the feature against abuse or limit the possible scope of mischief.  I understand that iterating using only static UCRs is a big pain, but I don't think this is an acceptable compromise.  IIRC, the aim of "sum when templates" was to allow for at least some use in dynamic UCRs by encoding the SQL as a template.  This means the actual SQL expressions must go through code review, but the rest of the UCR can be dynamic.  Is that an option here?

mkangia commented:
Thanks for raising that @esoergel 

Currently it's not enabled for any domain on prod and only enabled for handful of leftover icds domains and a new one that we added now for a project in india.

I will be happy to revert this now since I have moved all reports to static and then put the expressions as sum when templates in code and use them instead, so that we can continue using dynamic reports for testing, if this looks pressing or else add a warning on the toggle to NOT enable it for any domain while I do that.

mkangia commented:
To add, the following line is already present on the feature flag

> This feature flag was created for one specific project. Please don't enable it for any other projects. This is NOT SUPPORTED outside of that project and may break other features.

esoergel commented:
Due to the potential exposure, I think reverting it would be most appropriate, though I don't think an off-schedule deploy is necessary.

mkangia commented:
sure @esoergel . I will try to revert and add the custom templates before the next deploy so that no one is blocked.
thanks for raising this.
